src/borg/testsuite/archiver.py

```diff
@@ -3603,6 +3603,39 @@ def test_recovery_from_deleted_repo_nonce(self):
         self.cmd('create', self.repository_location + '::test2', 'input')
         assert os.path.exists(nonce)
 
+    def test_init_defaults_to_argon2(self):
+        """https://github.com/borgbackup/borg/issues/747#issuecomment-1076160401"""
+        self.cmd('init', '--encryption=repokey', self.repository_location)
+        with Repository(self.repository_path) as repository:
+            key = msgpack.unpackb(a2b_base64(repository.load_key()))
+        assert key[b'algorithm'] == b'argon2 aes256-ctr hmac-sha256'
+
+    def test_init_with_explicit_key_algorithm(self):
+        """https://github.com/borgbackup/borg/issues/747#issuecomment-1076160401"""
+        self.cmd('init', '--encryption=repokey', '--key-algorithm', 'pbkdf2', self.repository_location)
+        with Repository(self.repository_path) as repository:
+            key = msgpack.unpackb(a2b_base64(repository.load_key()))
+        assert key[b'algorithm'] == b'sha256'
+
+    def test_change_passphrase_does_not_change_algorithm(self):
+        self.cmd('init', '--encryption=repokey', '--key-algorithm', 'argon2', self.repository_location)
+        os.environ['BORG_NEW_PASSPHRASE'] = 'newpassphrase'
+
+        self.cmd('key', 'change-passphrase', self.repository_location)
+
+        with Repository(self.repository_path) as repository:
+            key = msgpack.unpackb(a2b_base64(repository.load_key()))
+            assert key[b'algorithm'] == b'argon2 aes256-ctr hmac-sha256'
```

Comment:
well, guess this test should start from pbkdf2 key, because:
- this is what people usually have now
- we would change algorithm pbkdf2 -> argon2 (but not argon2 -> pbkdf2) **if** we would do automated upgrades on change passphrase. so if you want to make sure here that we do **not** upgrade, you need to use the constellation that would actually get upgraded.